source/binding/Statements.cpp

```diff
@@ -528,10 +534,26 @@ bool StatementList::verifyConstantImpl(EvalContext& context) const {
     return true;
 }
 
+void StatementList::serializeTo(ASTSerializer& serializer) const {
+    serializer.startArray("list");
+    for (auto const& stmt : list) {
+        if (stmt)
```

Comment:
You don't need to check list items for null, we basically don't ever put null items into lists. Also consider just using plain `auto` when the item is a pointer. There's no reason to take a reference to the pointer.

Reply:
Sorry haven't touched raw pointers for a while. I will refactor the code.